protocol/test/forge/v2/strategies/AbstractStrategy.t.sol

```diff
@@ -332,17 +339,160 @@ contract AbstractStrategyTestBalances is AbstractStrategyTestBase {
+            debtCeiling[0] = ITempleStrategy.AssetBalance(address(dai), 333e18);
+            debtCeiling[1] = ITempleStrategy.AssetBalance(address(temple), 66e18);
+            debtCeiling[2] = ITempleStrategy.AssetBalance(address(weth), 99e18);
+            trv.addStrategy(address(strategy), -123, debtCeiling);
+        }
+
+        vm.stopPrank();
+    }
+
+    function approxInterest(uint256 principal, uint96 rate, uint256 age) internal pure returns (uint256) {
+        // Approxmiate as P * (1 + r/365 days)^(age)
+        // ie compounding every 1 second (almost but not quite continuous)
+        uint256 onePlusRate = uint256(rate / 365 days + 1e18);
+
+        return ud(principal).mul(ud(onePlusRate).powu(age)).unwrap();
+    }
+
+    function test_multiAsset() public {
+        vm.startPrank(executor);
+        {
+            strategy.borrow(dai, 100e18);
+            strategy.borrow(temple, 50e18);
+            strategy.borrow(weth, 2e18);
+        }
+
+        {
+            assertEq(dai.balanceOf(address(strategy)), 100e18);
+            assertEq(dUSD.balanceOf(address(strategy)), 100e18);
+
+            assertEq(temple.balanceOf(address(strategy)), 50e18);
+            assertEq(dTEMPLE.balanceOf(address(strategy)), 50e18);
+
+            assertEq(weth.balanceOf(address(strategy)), 2e18);
+            assertEq(dETH.balanceOf(address(strategy)), 2e18);
+        }
+
+        vm.warp(block.timestamp + 365 days);
+
+        {
+            // 1% interest for dUSD
+            assertEq(dai.balanceOf(address(strategy)), 100e18);
+            assertApproxEqRel(
+                dUSD.balanceOf(address(strategy)),
+                approxInterest(100e18, uint96(DEFAULT_BASE_INTEREST), 365 days),
+                1e8
+            );
+
+            // No interest for dTEMPLE
+            assertEq(temple.balanceOf(address(strategy)), 50e18);
+            assertEq(dTEMPLE.balanceOf(address(strategy)), 50e18);
+
+            // 4% interest for dUSD
+            assertEq(weth.balanceOf(address(strategy)), 2e18);
+            assertApproxEqRel(
+                dETH.balanceOf(address(strategy)),
+                approxInterest(2e18, uint96(ETH_BASE_INTEREST), 365 days), 
+                1e8
+            );
+        }
+
+        // Repay all the principal
+        {
+            strategy.repay(dai, 100e18);
+            strategy.repay(temple, 50e18);
+            strategy.repay(weth, 2e18);
+        }
+
+        {
+            // 1% interest for dUSD
+            assertEq(dai.balanceOf(address(strategy)), 0);
+            assertApproxEqRel(
+                dUSD.balanceOf(address(strategy)),
+                approxInterest(100e18, uint96(DEFAULT_BASE_INTEREST), 365 days) - 100e18,
+                1e10
+            );
+
+            // No interest for dTEMPLE
+            assertEq(temple.balanceOf(address(strategy)), 0);
+            assertEq(dTEMPLE.balanceOf(address(strategy)), 0);
+
+            // 4% interest for dUSD
```

Comment:
same

Reply:
Fixed in https://github.com/TempleDAO/temple/pull/814/commits/3933dcacf45c04e2bb680b56d4921a0d6b6ba65a